scripts/backCompatChecker.py

```diff
@@ -85,21 +93,27 @@ def findExtension():
             if file.endswith('.whl'):
                 return os.path.join(p, file)
 
+
 newExtensionLocation = findExtension()
+print('Install extension (loaded from current code). Wheel path - {}'.format(newExtensionLocation))
 subprocess.run(['az', 'extension', 'add', '--source', newExtensionLocation, '-y'], shell=True, stdout=subprocess.PIPE)
 
+# Check the installed extensions
+subprocess.run(['az', 'extension', 'list'], shell=True)
+
 # get a set of old commands, we are not reusing the set from ext because we want to keep this clean
 oldCommands = []
 for oldArgument in oldArguments:
     if oldArgument.command not in ignoreCommands:
-        if not any(oldArgument.command in s for s in oldCommands):
```

Comment:
The check fails for command = 'devops security permission reset' list containing 'devops security permission reset-all' and thus all params of command are being detected as missing. 

Still not sure how it started failing without any code change.. did the python behavior change?

Also since the preview mechanism is changed these commands are the ones which were previously not even getting checked for back compat but are now because the 'PREVIEW' check in this script will not work.

Reply:
Rest of the changes in the file are only to improve the debugging from logs.

Reply:
I agree, how did it fail all of a sudden since, the python version(3.7) and azure-cli version(2.0.75) looks same on recent and earlier pipelines. 
But somehow, the command PREVIEW tags are still present in backCompactChecker script logs, even though, azure-cli version 2.0.75 already has the extension PREVIEW feature inbuilt.

Reply:
> I agree, how did it fail all of a sudden since, the python version(3.7) and azure-cli version(2.0.75) looks same on recent and earlier pipelines.
> But somehow, the command PREVIEW tags are still present in backCompactChecker script logs, even though, azure-cli version 2.0.75 already has the extension PREVIEW feature inbuilt.

That is because with new preview mechanism the text contains 'preview' and not 'PREVIEW' which is what we check in the script to exclude those commands.

Reply:
Updating here for completing the thread... As discussed offline.. 

The issue here was were running this test against the already installed version of extension in the agent which was 0.12.0 and preview commands were not getting tested because they are excluded by the script. 
**FIX 1: This is fixed by adding a extension remove command at start of script.**

The agent got updated and the extension version 0.13.0 was available where the PREVIEW check failed to block the commands and this started failing due to a code issue which was always there which returns positive match for list membership even on substring match. 
**FIX 2: Check for preview according to new code.**
**FIX 3: Substring match issue in list membership check.**